assets/js/modules/analytics-4/datastore/webdatastreams.js

```diff
@@ -134,6 +173,37 @@ const baseSelectors = {
 	getWebDataStreams( state, propertyID ) {
 		return state.webdatastreams[ propertyID ];
 	},
+
+	/**
+	 * Gets matched web data stream for selected property.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state      Data store's state.
+	 * @param {string} propertyID The GA4 property ID to find matched web data stream.
+	 * @return {(Object|null|undefined)} A web data stream object if found, otherwise null; `undefined` if web data streams are not loaded.
+	 */
+	getMatchingWebDataStream: createRegistrySelector( ( select ) => ( state, propertyID ) => {
+		const datastreams = select( STORE_NAME ).getWebDataStreams( propertyID );
+		if ( datastreams === undefined ) {
+			return undefined;
+		}
+
+		if ( Array.isArray( datastreams ) ) {
```

Comment:
`getWebDataStreams` only returns `Array|undefined` so this would be an unnecessary check. I would rather we enforce the type when receiving the data into state than not trusting our own APIs.

Reply:
Updated.